lib/stoplight/default.rb

```diff
@@ -2,16 +2,15 @@
 
 module Stoplight
   module Default
-    WHITELISTED_ERRORS = [
-      NoMemoryError,
-      ScriptError,
-      SecurityError,
-      SignalException,
-      SystemExit,
-      SystemStackError
-    ].freeze
-
-    BLACKLISTED_ERRORS = [].freeze
+    ERROR_HANDLER = lambda do |error|
+      classes = [
+        Interrupt,
+        NoMemoryError,
+        SignalException,
+        SystemExit
+      ]
+      raise error if classes.any? { |klass| error.is_a?(klass) }
+    end
```

Comment:
I understand your drive for simplicity, which I can see very well in this well factored project, and I think your way is better ascetically. I like the idea of removing the whitelists and blacklists completely.

Never the less this approach could lead to an API that's too raw, giving user the ability to shoot them self's in the foot. 

1. A user provided error handler will have to know about the special classes in the default handler, in some cases, in order not to rescue critical errors.
2. There could be system side effect with having to re-raise the error, especially since there are such critical errors.

I think there are some very smart people working over at RSpec and would prefer their `Module::===` method for validating what errors to rescue.
https://github.com/rspec/rspec-support/blob/master/lib/rspec/support.rb#L122

It may also make for a better `Proc` API to return `true` if the user want Stoplight to handle the issue and `false` for Stoplight to ignore it. This would take a little meta programming to create a dynamic module that takes the user supplied handler.

Sorry for changing my mind so many times.